esp_bool_parser/bool_parser.py

```diff
@@ -94,8 +95,7 @@ def get_value(self, target: str, config_name: str) -> t.Any:
         if self.attr == 'IDF_VERSION_PATCH':
             return IDF_VERSION_PATCH
 
-        if self.attr == 'CONFIG_NAME':
-            return config_name
+        from .soc_header import SOC_HEADERS
```

Comment:
How about we prioritize loading the environment variables? so if I rely on one env var defined locally, there's no need to check and load the IDF_PATH (which may not exist)

The original comment:

> for non-keyword cap words, check if it is defined in the environment variables

no idea why I put it here... I can't think of a real risk about it.

Reply:
overall the sequence could be

- additional_attr
- IDF_TARGET
- CONFIG_NAME
- env var
- IDF_VERSION(_XXX) - lazy-load
- SOC_CAPS - lazy-load